src/main/scala/com/beachape/filemanagement/FileSystemWatchActor.scala

```diff
@@ -0,0 +1,62 @@
+package com.beachape
+
+/**
+ * Created by arunavs on 9/27/16.
+ *
+ * This is a simple actor that can be used directly to send a message
+ * whenever a dir is modified.
+ *
+ * It sends the changed dir message whenever changes happen to the
+ * directory.
+ */
+
+import java.nio.file.{ Path, Paths }
+
+import com.beachape.filemanagement.Messages.RegisterCallback
+import com.beachape.filemanagement.MonitorActor
+import com.beachape.filemanagement.RegistryTypes.Callback
+import java.nio.file.StandardWatchEventKinds._
+
+import akka.actor.ActorRef
+import com.beachape.FileSystemWatchActor.ChangedDir
+
+class FileSystemWatchActor(dirs: List[String], sendToActor: ActorRef) extends MonitorActor {
```

Comment:
Thinking about it again, the naming of this actor is not very descriptive; it's in essence a message forwarding actor..